src/core/particle_data.cpp

```diff
@@ -76,6 +76,7 @@ void add_id_to_type_map(int part_id, int type);
 
 int max_seen_particle = -1;
 int n_part = 0;
+bool has_swimming_particles = false;
```

Comment:
What "has" swimming particles? Maybe I'm pedantic, but a name like ```swimming_particles_exist``` would make more sense in my opinion.